### PR TITLE
need to add spdlog to cmake

### PR DESCRIPTION
**Pull request by joequant**
need to add spdlog to cmake in order to build

### CODE REVIEW

File: var/spack/repos/builtin/packages/prmon/package.py [hunk 1]
```diff
@@ -32,6 +32,8 @@ class Prmon(CMakePackage):
     depends_on('py-matplotlib', type="run", when="+plot")
     depends_on('spdlog', when='@3.0.0:')
 
+    patch('cmake.patch')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/prmon/package.py:35]: Does the patch apply to every version we currently have?

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by vvolkl** [comment]
Hmm, I see that prmon assumes that spdlog is set up as a submodule and does not find it in cmake because of that. Let me see if we can get a solution accepted upstream that works with both external and submodule-ed spdlog.

**Comment by chuckatkins** [comment]
> Hmm, I see that prmon assumes that spdlog is set up as a submodule and does not find it in cmake because of that. Let me see if we can get a solution accepted upstream that works with both external and submodule-ed spdlog.

I happened to see this PR in passing this morning and wanted to take a quick shot at it so I submitted a fix upstream:
https://github.com/HSF/prmon/pull/220

The PR allows both `nlohmann_json` and `spdlog` to be use as either internal or external dependencies and I tested it applying as a patch back to `v3.0.0`.  That would allow this to then be:

```diff
--- a/var/spack/repos/builtin/packages/prmon/package.py
+++ b/var/spack/repos/builtin/packages/prmon/package.py
@@ -32,11 +32,22 @@ class Prmon(CMakePackage):
     depends_on('py-matplotlib', type="run", when="+plot")
     depends_on('spdlog', when='@3.0.0:')
 
+    # See https://github.com/HSF/prmon/pull/220
+    patch('pr220-external-deps.patch', when='@3:')
+
     def cmake_args(self):
         args = [
             # googletest is fetched and built on the fly
             self.define("BUILD_GTESTS", self.run_tests),
+            self.define("BUILD_TESTING", self.run_tests),
         ]
+
+        if '@3:' in self.spec:
+            args.extend([
+                self.define('USE_EXTERNAL_NLOHMANN_JSON', True),
+                self.define('USE_EXTERNAL_SPDLOG', True),
+            ])
+
         return args
 
     def check(self):
```

Don't know how responsive upstream is to these things, but the pr should address the issue.

**Comment by vvolkl** [comment]
Superseded by #30979

**Pull request closed without merging**